File: ruby/raindrops/raindrops.rb
class Raindrops
  class << self
    def convert(value)
      new(value).sound
    end
  end

  attr_reader :value
  
  RAINDROPS = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong'
  }.freeze

  def initialize(value)
    @value = value
  end

  def factors
    RAINDROPS.map { _2 if (@value % _1).zero? }.compact.join
  end

  def sound
    return value.to_s if factors.empty?

    factors
  end
end

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

func Convert(value int) string{
	raindrops := map[int]string{
		3: "Pling",
		5: "Plang",
		7: "Plong",
	}
	
	var result string

	for i := range raindrops {
		if value % i == 0 {
			result += raindrops[i]
		}
	}

	if len(result) == 0{
		return strconv.Itoa(value)
	} 

	return result
}


